elevate-cpanel

```diff
@@ -348,17 +359,30 @@ sub run ($self) {
 
 sub _parse_opt_upgrade_to ($self) {
 
+    if ( Cpanel::OS::distro() eq 'cloudlinux' && Cpanel::OS::major() == 7  ) {
+        INFO("Running on CloudLinux 7, upgrade target set to CloudLinux 8.");
+        $self->{upgrade_to} = UPGRADE_TO_CLOUDLINUX;
+
+        if ( defined $self->getopt('upgrade-to') ) {
+            die qq[--upgrade-to option is not supported on CloudLinux systems\n];
+        }
+    }
+
     return unless my $flavor = $self->getopt('upgrade-to');
 
     if ( !defined $self->getopt('start') && !defined $self->getopt('check') ) {
         die qq[--upgrade-to option is only supported with --start or --check\n];
     }
     $flavor = lc $flavor;
     if ( $flavor !~ m{^(?:almalinux|alma|rocky)$} ) {
-        die qq[Invalid --upgrade_to value '$flavor'. Only 'almalinux' or 'rocky' are supported.\n];
+        die qq[Invalid --upgrade_to value '$flavor'. Only 'almalinux' and 'rocky' are supported.\n];
     }
 
-    $self->{upgrade_to} = $flavor eq 'rocky' ? UPGRADE_TO_ROCKY : UPGRADE_TO_ALMALINUX;
+    if ($flavor eq 'rocky') {
+        $self->{upgrade_to} = UPGRADE_TO_ROCKY;
+    } else {
```

Comment:
elsif? Seems like you're going to cause this to be alma on cloud?

Reply:
This essentially preserves the behavior of the ternary operator that was used beforehand.
In what cases would we get an upgrade to Alma where we didn't want Alma? The above `flavor` check won't let anything other than alma/rocky through, and the function will return earlier if `upgrade-to` is empty.